Makefile

```diff
@@ -161,8 +161,31 @@ sub-image-%:
 	@$(MAKE) ARCH=$* image
 
 image: ## make the image for a single ARCH
-	docker buildx build --load -t $(BUILD_IMAGE):latest-$(ARCH) -f Dockerfile --platform $(OS)/$(ARCH) .
-	echo "Done. image is at $(BUILD_IMAGE):latest-$(ARCH)"
+	docker buildx build --load -t $(BUILD_IMAGE):$(BUILD_TAG)-$(ARCH) -f Dockerfile --platform $(OS)/$(ARCH) .
+	echo "Done. image is at $(BUILD_IMAGE):$(BUILD_TAG)-$(ARCH)"
+
+image-all-push: $(addprefix sub-image-push-, $(ARCHES)) ## Push all built images.
+sub-image-push-%:
+	@$(MAKE) ARCH=$* image-push
+
+image-push: ## Push image to registry for a single ARCH.
```

Comment:
1. Is there any reason not to call this just `push`? The names of the targets are getting mighty long and confusing.
2. Shouldn't this have `image` as a dependency?

Reply:
>Is there any reason not to call this just push? The names of the targets are getting mighty long and confusing.

Well, it puts all Docker image related targets into a `image-` prefix, which I think gives nice grouping. perhaps `sub-image-push` could be changed to `image-sub-push` or `image-push-sub` ? Or Maybe `image-push-arch` ?

>Shouldn't this have image as a dependency?

Yes :+1:

Reply:
> it puts all Docker image related targets into a image- prefix, which I think gives nice grouping. perhaps sub-image-push could be changed to image-sub-push or image-push-sub ? Or Maybe image-push-arch ?

I get it, but I don't think it is necessary. It buys us unnecessary flexibility at the price of additional complexity. What else are we going to `push` from here? What else are we going to do with `manifest`?

Reply:
`manifest` might be related to Kubernetes manifests, which is more likely than Docker manifest I think. But yeah, I don't insist on that, let me change it.